packages/aws-cdk/lib/docker.ts

```diff
@@ -78,33 +64,33 @@ export async function prepareContainerAsset(assemblyDir: string,
       baseCommand.push('--file', asset.file);
     }
 
-    const command = ci
-      ? [...baseCommand, '--cache-from', latest] // This does not fail if latest is not available
-      : baseCommand;
-    await shell(command);
+    await shell(baseCommand);
 
     // Login and push
-    if (!loggedIn) { // We could be already logged in if in CI
-      await dockerLogin(toolkitInfo);
-      loggedIn = true;
-    }
+    await dockerLogin(toolkitInfo);
 
     // There's no way to make this quiet, so we can't use a PleaseHold. Print a header message.
     print(` ⌛ Pushing Docker image for ${contextPath}; this may take a while.`);
-    await shell(['docker', 'push', latest]);
+    await shell(['docker', 'push', imageUri]);
     debug(` 👑  Docker image for ${contextPath} pushed.`);
 
-    // Get the (single) repo-digest for latest, which'll be <ecr.repositoryUrl>@sha256:<repoImageSha256>
-    const repoDigests = (await shell(['docker', 'image', 'inspect', latest, '--format', '{{range .RepoDigests}}{{.}}|{{end}}'])).trim();
-    const requiredPrefix = `${ecr.repositoryUri}@sha256:`;
-    const repoDigest = repoDigests.split('|').find(digest => digest.startsWith(requiredPrefix));
-    if (!repoDigest) {
-      throw new Error(`Unable to identify repository digest (none starts with ${requiredPrefix}) in:\n${repoDigests}`);
+    // backwards compatibility with pre 1.21.0, wire imageNameParameter to the actual image name
+    if (asset.imageNameParameter) {
+      // Get the (single) repo-digest for latest, which'll be <ecr.repositoryUrl>@sha256:<repoImageSha256>
+      const repoDigests = (await shell(['docker', 'image', 'inspect', imageUri, '--format', '{{range .RepoDigests}}{{.}}|{{end}}'])).trim();
```

Comment:
Shouldn't this whole calculation just be replaced with `imageUri`? (Will only work if we don't use `:latest` though)

Reply:
Previous versions currently rely on this image name format as it is passed through a parameter, which includes the image's SHA-256 digest. If we just pass the `imageUri` here it will break current code.

Reply:
Left current code path in tact, with support for `imageTag`.